File: Data Structures & Algorithms/baseball-game/submission-1.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        st = list()
        total = 0
        for i in operations:
            if i == "+":
                st.append(st[-1] + st[-2])
                total += st[-1]
            elif i == "D":
                st.append(2 * st[-1])
                total += st[-1]
            elif i == "C":
                total -= st[-1]
                st.pop()
            else:
                st.append(int(i))
                total += st[-1]
        return total

File: Data Structures & Algorithms/decode-string/submission-1.cpp
class Solution {
    static string repeat(const std::string& s, int n) {
        std::string result;
        result.reserve(s.length() * n); // Pre-allocate memory for efficiency
        for (int i = 0; i < n; ++i) {
            result += s; // Or result.append(s);
        }
        return result;
    }
public:
    string decodeString(string s) {
        stack<int> st;
        stack<string> sst;
        const int n = static_cast<int>(s.length());
        string temp;
        for (int i = 0; i < n; i++) {
            if (isdigit(s[i])) {
                string times;
                while (isdigit(s[i]) && i < n) {
                    times.push_back(s[i]);
                    i++;
                }
                i--;
                st.push(stoi(times));
                sst.push(temp);
                temp = "";
                i++;
            } else if (s[i] == ']') {
                const int t = st.top();
                st.pop();
                temp = repeat(temp, t);
                sst.top() += temp;
                temp = sst.top();
                sst.pop();
            } else {
                temp.push_back(s[i]);
            }
        }
        return temp;
    }

};